python/src/aiconfig/editor/server/async_queue_iterator.py

```diff
@@ -0,0 +1,40 @@
+"""Queue iterator for streaming. Can only process strings for now
+but in future will try to make it generic"""
+import asyncio
+# from typing import Generic, TypeVar
+
+# # TODO: Add generic typing for queue items
+# # (couldn't get sentinel value to work with generics)
+# T = TypeVar('T')
+STOP_STREAMING_SIGNAL = object() #sentinel value to indicate end of stream
+
+class AsyncQueueIterator():
+    """In order to support text streaming, we need to store
+    the output in a queue and iterate over those values. A lot of this was
+    inspired by HuggingFace's TextIteratorStreamer object:
+    
+    I know I can just use a queue directly in the callsite with 
+    `iter(queue.get, None)`, but having a class makes it easier to manage 
+    and abstracts it a bit more.
+    """
+    def __init__(self):
+        self.q = asyncio.Queue()
+        self.stop_signal = STOP_STREAMING_SIGNAL
+
+    def __aiter__(self):
+        return self
+
+    def __next__(self):
+        value = await self.q.get()
+        if value == self.stop_signal:
+            raise StopIteration()
+        else:
+            return value
```

Comment:
Currently learning how to convert iterators --> async iterators: https://medium.com/geekculture/asynchronous-iterators-in-python-fdf55198287d

Need to refactor this